src/content/docs/en/guides/middleware.mdx

```diff
@@ -19,11 +19,12 @@ Middleware is available in both SSG and SSR Astro projects.
 
     ```js title="src/middleware.js"
     export function onRequest ({ locals, request }, next) {
-        // intercept response data from a request
-        // optionally, transform the response by modifying `locals`
+        // intercept data from a request
+        // optionally, share data with the rendering process by modifying `locals`
         locals.title = "New title";
 
         // return a Response or the result of calling `next()`
+        // optionally, transform the response by modifying the result
```

Comment:
This doesn't feel well-placed to me, and in fact I don't even think it's necessary at all, if the we get the above code comment just right. The existing comment here is meant to highlight that you have to RETURN something, either `next()`  or a `Response` directly. I think this statement here only adds extra information gets (literally!) in the way of the point it's trying to make.

Is this also in fact correct? is the modifying done after what is returned in this function? Isn't returning *something* the final step here? So, if there are in fact two code comments, it seems to me like the one about what is returned by the middleware should go last.

I can think of some options off the top of my head, depending on what's most accurate, that I think feel more *helpful* to the reader:

1. change the existing statement to include the idea that the data, potentially modified, is passed along in the thing that's returned. e.g. "return a Response or the result of calling `next`, passing along `context.locals`"
2. swap the order of these, and reword the first one include information you believe is truly useful to include.
3. delete this, because it probably doesn't need to be spelled out HERE that the new info is available if the code comment above clearly explains the purpose behind transforming some data to mutate and pass along `context.locals`.

Reply:
The distinction is that:
- You can modify the `locals` to affect how the page/endpoint _will_ behave
- You can modify the return of `next` _after_ the page/endpoint has already completed

There are examples of this in the same page. For the first case, there is an example of changing the title inside `locals` allowing the page to render it differently. For the second, there is an example of redacting sensitive information by replacing segments of the resulting HTML.

Reply:
Taking another look at it, I agree that it gets in the way of the most essential information. Since there is already another section explaining about modifying the result of `next` I'll remove this comment.

Reply:
That works for me!